sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -587,7 +601,8 @@ object DataSource extends Logging {
                 if (provider1.toLowerCase(Locale.ROOT) == "orc" ||
```

Comment:
`provider1` can't be "orc" anymore. It can be only `classOf[OrcFileFormat].getCanonicalName` or `"org.apache.spark.sql.hive.orc.OrcFileFormat"`.

Reply:
Yep. I'll remove this, `provider1.toLowerCase(Locale.ROOT) == "orc" ||`.